src/encoded/audit/experiment.py

```diff
@@ -3932,6 +3932,75 @@ def audit_experiment_nih_institutional_certification(value, system, excluded_typ
         yield AuditFailure('missing nih_institutional_certification', detail, level='ERROR')
 
 
+def audit_experiment_eclip_queried_RNP_size_range(value, system, excluded_types):
+    '''
+    Check if libraries of eCLIP experiment and its control have matching queried_RNP_size_range.
+    '''
+    if value.get('assay_term_name') != 'eCLIP':
+        return
+
+    experiment_size_range = set()
+    control_size_range = set()
+    details = []
+    control_accessions = []
+
+    for rep in value['replicates']:
+        if rep.get('status') not in excluded_types and 'library' in rep:
+            lib = rep.get('library', {})
+            if 'queried_RNP_size_range' in lib:
+                experiment_size_range.add(lib['queried_RNP_size_range'])
+            else:
+                details.append(
+                    'Library {} is missing specification of queried_RNP_size_range.'.format(
+                        audit_link(path_to_text(lib['@id']), lib['@id'])
+                    )
+                )
+
+    for control in value.get('possible_controls'):
+        control_accessions.append(audit_link(path_to_text(control['@id']), control['@id']))
+        for rep in control['replicates']:
+            if rep.get('status') not in excluded_types and 'library' in rep:
+                lib = rep.get('library', {})
+                if 'queried_RNP_size_range' in lib:
+                    control_size_range.add(lib['queried_RNP_size_range'])
+                else:
+                    details.append(
+                        'Library {} is missing specification of queried_RNP_size_range.'.format(
+                            audit_link(path_to_text(lib['@id']), lib['@id'])
+                        )
+                    )
+
+    if details:
+        for detail in details:
+            yield AuditFailure('missing queried_RNP_size_range', detail, level='ERROR')
+        return
+
+    if len(experiment_size_range) > 1:
+        detail = 'Libraries of experiment {} have mixed queried_RNP_size_range values of {}.'.format(
+            audit_link(path_to_text(value['@id']), value['@id']),
+            ', '.join(experiment_size_range)
+        )
+        yield AuditFailure('mixed queried_RNP_size_range', detail, level='ERROR')
+
+    if len(control_size_range) > 1:
+        detail = 'Libraries of control experiment {} have mixed queried_RNP_size_range values of {}.'.format(
+            ', '.join(control_accessions),
+            ', '.join(control_size_range)
+        )
+        yield AuditFailure('mixed queried_RNP_size_range', detail, level='ERROR')
+
+    if experiment_size_range != control_size_range and value['possible_controls']:
```

Comment:
I think only if the size ranges in experiment and in control are "OK" you should continue to the comparison between them, otherwise, you are potentially comparing multiple ranges in control to a range in the experiment. In addition to that, the Detail message in the "inconsistent queried .." audit should be not sets but the string values of the size_ranges.